locations/commands/insights.py

```diff
@@ -127,6 +146,69 @@ def check_wikidata_codes(self, args):
         self.show_counter("NSI MISSING WIKIDATA CODE:", spider_nsi_missing_counter)
 
+    def analyze_atp_nsi_osm(self, args, outfile):
+        """
+        Trawl ATM, NSI and OSM for per wikidata code information.
+        :param args: ATP output GeoJSON files / directories to load
+        :param outfile: JSON result file name to write
+        """
+
+        def lookup_code(wikidata_code):
+            # Return record for a wikidata code, adding it to dict if not already present.
+            if record := wikidata_dict.get(wikidata_code):
+                return record
+            record = {
+                "code": wikidata_code,
+                "osm_count": 0,
+                "nsi_brand": None,
+                "nsi_description": None,
+                "atp_count": None,
+                "atp_brand": None,
+            }
+            wikidata_dict[wikidata_code] = record
+            return record
+
+        # A dict keyed by wikidata code.
+        wikidata_dict = {}
+
+        # First data set to merge into the output table is wikidata tag count info from OSM.
+        osm_url = "https://taginfo.openstreetmap.org/api/4/key/values?key=brand%3Awikidata&filter=all&lang=en&sortname=count&sortorder=desc&page=1&rp=000&qtype=value"
+        response = requests.get(osm_url)
+        if not response.status_code == 200:
+            raise Exception("Failed to load OSM wikidata tag statistics")
+        for r in response.json()["data"]:
+            lookup_code(r["value"])["osm_count"] = r["count"]
+
+        # Now load each wikidata entry in the NSI dataset and merge into our wikidata table.
+        nsi = NSI()
+        for k, v in nsi.iter_wikidata():
+            r = lookup_code(k)
+            r["nsi_brand"] = v.get("label", "NO NSI LABEL!")
+            r["nsi_description"] = v.get("description", "NO NSI DESC!")
+
+        # TODO: Could go through ATP spiders themselves looking for Q-codes. Add an atp_count=-1
+        #       which would be written over if a matching POI had been scraped by the code below.
+        #       If not then that would be a possible problem to highlight.
+
+        # Walk through the referenced ATP downloads, if they have wikidata codes then update our output table.
+        for feature in iter_features(args):
+            brand_wikidata = feature["properties"].get("brand:wikidata")
+            brand = feature["properties"].get("brand")
+            if not brand_wikidata:
+                continue
+            r = lookup_code(brand_wikidata)
+            count = r.get("atp_count")
+            if not count:
+                count = 0
+            r["atp_count"] = count + 1
+            if brand:
+                r["atp_brand"] = brand
+
+        # Write a JSON format output file which is datatables friendly.
+        for_datatables = {"data": list(wikidata_dict.values())}
+        with open(outfile, "w") as f:
+            f.write(json.dumps(for_datatables))
```

Comment:
You don't have to json.dumps first.

```suggestion
            json.dump(for_datatables, f)
```